File: examples/ci/jenkins/cenario_1/pact-consumer-sample/src/main/java/br/com/vinirib/pact/consumer/client/integration/account/service/impl/AccountIntegrationServiceImpl.java
package br.com.vinirib.pact.consumer.client.integration.account.service.impl;

import br.com.vinirib.pact.consumer.client.dto.BalanceDTO;
import br.com.vinirib.pact.consumer.client.integration.account.service.AccountIntegrationService;
import java.util.Optional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class AccountIntegrationServiceImpl implements AccountIntegrationService {

    @Autowired
    private RestTemplate restTemplate;

    @Value("${integration.account.balance.url}")
    private String accountBalanceUrl;

    @Override
    public Optional<BalanceDTO> getBalance(Integer accountId) {
        final ResponseEntity<BalanceDTO> responseEntity = restTemplate.getForEntity(accountBalanceUrl, BalanceDTO.class, accountId);
        return Optional.ofNullable(responseEntity.getBody());
    }
}

File: examples/kotlin/one-consumer-one-provider/consumer/src/main/kotlin/br/com/zup/pact/consumer/integration/account/service/impl/AccountIntegrationServiceImpl.kt
package br.com.zup.pact.consumer.integration.account.service.impl

import br.com.zup.pact.consumer.dto.BalanceDTO
import br.com.zup.pact.consumer.integration.account.service.AccountIntegrationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import org.springframework.web.client.RestTemplate

@Service
class AccountIntegrationServiceImpl (
        @Value("\${integration.account.balance.url}") private val accountBalanceUrl: String,
        @Autowired private val restTemplate: RestTemplate
): AccountIntegrationService {

    override fun getBalance(accountId: Int): BalanceDTO? {
        val responseEntity: ResponseEntity<BalanceDTO> = restTemplate.getForEntity(
                accountBalanceUrl,
                BalanceDTO::class.java,
                accountId
        )
        return responseEntity.body
    }
}